main.tf

```diff
@@ -1 +1,98 @@
+  location            = var.virtual_machine_config.location
+  resource_group_name = var.resource_group_name
+  dns_servers         = var.nic_config.dns_servers
+
+  ip_configuration {
+    name                          = local.nic.ip_config_name
+    subnet_id                     = var.subnet.id
+    private_ip_address_allocation = var.nic_config.private_ip == null ? "Dynamic" : "Static"
+    private_ip_address            = var.nic_config.private_ip
+    public_ip_address_id          = var.public_ip_config.enabled ? azurerm_public_ip.this[0].id : null
+  }
+
+  lifecycle {
+    ignore_changes = [
+      tags
+    ]
+  }
+}
+
+resource "azurerm_network_interface_security_group_association" "this" {
+  count                     = length(var.nic_config.nsg_name) > 0 ? 1 : 0
+  network_interface_id      = azurerm_network_interface.this.id
+  network_security_group_id = data.azurerm_network_security_group.this[0].id
+}
+
+resource "azurerm_linux_virtual_machine" "this" {
+  name                            = local.virtual_machine.name
+  computer_name                   = var.virtual_machine_config.hostname
+  location                        = var.virtual_machine_config.location
+  resource_group_name             = var.resource_group_name
+  size                            = var.virtual_machine_config.size
+  provision_vm_agent              = true
+  admin_username                  = var.virtual_machine_config.admin_username
+  admin_password                  = var.admin_password
+  disable_password_authentication = length(var.admin_password) > 0 && length(var.public_key) == 0 ? false : true
+
+  dynamic "admin_ssh_key" {
+    for_each = length(var.public_key) > 0 ? [1] : []
+    content {
+      username   = var.virtual_machine_config.admin_username
+      public_key = var.public_key
+    }
+  }
+
+  network_interface_ids = [
+    azurerm_network_interface.this.id,
+  ]
+
+  os_disk {
+    name                      = var.virtual_machine_config.os_disk_name
+    caching                   = var.virtual_machine_config.os_disk_caching
+    disk_size_gb              = var.virtual_machine_config.os_disk_size_gb
+    storage_account_type      = var.virtual_machine_config.os_disk_storage_type    
+    write_accelerator_enabled = var.virtual_machine_config.write_accelerator_enabled
+  }
+
+  source_image_reference {
+    publisher = var.virtual_machine_config.os_publisher
+    offer     = var.virtual_machine_config.os_offer
+    sku       = var.virtual_machine_config.os_sku
+    version   = var.virtual_machine_config.os_version
+  }
+
+  availability_set_id = length(var.virtual_machine_config.availability_set_id) > 0 ? var.virtual_machine_config.availability_set_id : null
+  zone                = length(var.virtual_machine_config.zone) > 0 && var.virtual_machine_config.availability_set_id == null ? var.virtual_machine_config.zone : null
+  tags                = merge(var.virtual_machine_config.tags, {"Severity Group Monthly" = var.severity_group})
```

Comment:
As it turns out we have another mandatory tag for VMs called "Update allowed". Please check the windows VM module and copy from there so it defaults to "yes". See https://github.com/qbeyond/terraform-azurerm-windows-vm/blob/00a26a51f4d9f6a48377e87897a5f30e99ecb21e/locals.tf#L13 and https://github.com/qbeyond/terraform-azurerm-windows-vm/blob/00a26a51f4d9f6a48377e87897a5f30e99ecb21e/variables.tf#L99

Reply:
Fixed as windows module.